tools/fs/files.js

```diff
@@ -737,6 +745,32 @@ files.extractTarGz = function (buffer, destPath, options) {
   }
 };
 
+function waitFSToSync(dir) {
+	let waitToSync = true;
+
+	while (waitToSync) {
+		try {
+			// After a successful unlink operation, sometimes windows fs takes longer than expected to sync
+			// so readdir would still return already deleted files
+			const filesInTempDirArray = files.readdir(dir);
+
+			// If stat doesn't throw a known error, it means that any queued operations have already been performed
+			// and committed, so the list returned by readdir is the actual file list of the directory
+			filesInTempDirArray.forEach(file => files.stat(files.pathJoin(dir, file)));
+
+			waitToSync = false;
+		} catch (e) {
+			// EPERM is thrown eg after a successful unlink operation and readdir still returns it
+			// ENOENT is thrown if the file deletion is commited by fs between readdir and stat
+			if (e.code === 'EPERM' || e.code === 'ENOENT') {
+				waitToSync = true;
+			} else {
+				throw e;
+			}
+		}
+	}
+}
```

Comment:
Please use two spaces per tab (actual spaces, not `\t` characters) for indentation.